### ISSUE TITLE
Improvements to policy progress/validation reporting

### ISSUE THREAD

**Issue by joeduffy** [pulumi/pulumi-policy#90]
Per previous discussion

* Can we use ephemeral messages to include status updates?
    - Downloading
    - Installing
    - Checking s3-must-be-encrypted
    - Checking ...
    - Finished
* Errors don't need to span four lines; maybe
    ```
    mandatory: [s3-must-be-encrypted]: S3 buckets must be encrypted.
    aws:s3:Bucket "my-videos": expected 'true', got 'false'
    aws:s3:Bucket "my-other-videos": expected 'true', got 'false'
     ```
* Need summary of policy packs passed (otherwise, looks like I didn't even run them)

**Comment by hausdorff** [comment]
1. We should use ephemeral events, eventually—but they are currently emitted before the stack view is loaded. Since they currently block, this means that they complete before the stack view _ever_ has a chance to start, which means they end up being rendered sequentially anyway. So to get a good experience here, it seems like we'll have to do a bit of surgery to make everything async.
2. This seems reasonable, but I'm guessing we want all policy violations attributed to the resource, rather than the policy itself—this means that you can look at the report for one resource and see all its policy violations.
3. Yeah, we should definitely do this soon. We will have to wait until v1 launches though, due to the moratorium on new stuff going into the CLI.

**Comment by hausdorff** [comment]
Oh—the for (1) this is actually a duplicate of #58. So, once (2) and (3) close, we'll close this issue in favor of that one.

**Comment by joeduffy** [comment]
@hausdorff Let's try to make progress on this during M27. I personally find the current outputs quite confusing (don't know what ran, assert messages seem to always be semi incorrect, etc). Nailing the core UX and getting it to "lovable" status is important here.

**Comment by hausdorff** [comment]
Yes, let's fix. In particular, @joeduffy, what assert messages seem to be semi-incorrect? This used to be true, but should have been fixed—if I'd seen this recently I'd have fixed it immediately.

Some notes on the other stuff in this thread.

1. As I said above: when I looked into it last, installation occurs before the stack UI is set up, and it seemed like in that case ephemeral events are simply printed. This is not amazing, but it is a fair amount of work, and the other things here seem to be much higher impact. So I think we should revisit later.
1. We condensed errors down to two lines. One line seemed to be less clear, but open to suggestions about what to change to make that not true.
1. This is high importance, we'll fix this week.
